src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -815,50 +810,27 @@ protected virtual void Dispose(bool disposing)
             try
             {
                 _sender = new Ping();
-                _sender.PingCompleted += OnPingComplete;
 
                 timer?.Start();
-                _sender.SendAsync(targetAddress, timeout, buffer, pingOptions, this);
-                _pingComplete.Wait();
-                timer?.Stop();
-                _pingComplete.Reset();
-
-                if (_pingCompleteArgs == null)
-                {
-                    throw new PingException(string.Format(
-                        TestConnectionResources.NoPingResult,
-                        targetAddress,
-                        IPStatus.Unknown));
-                }
-
-                if (_pingCompleteArgs.Cancelled)
-                {
-                    // The only cancellation we have implemented is on pipeline stops via StopProcessing().
-                    throw new PipelineStoppedException();
-                }
-
-                if (_pingCompleteArgs.Error != null)
-                {
-                    throw new PingException(_pingCompleteArgs.Error.Message, _pingCompleteArgs.Error);
-                }
-
-                return _pingCompleteArgs.Reply;
+                return _sender.SendPingAsync(targetAddress, timeout, buffer, pingOptions).GetAwaiter().GetResult();
+            }
+            catch (PingException ex) when (ex.InnerException is TaskCanceledException)
+            {
+                // The only cancellation we have implemented is on pipeline stops via StopProcessing().
+                throw new PipelineStoppedException();
+            }
+            catch (Exception ex)
+            {
+                throw new PingException(ex.Message, ex);
```

Comment:
Since the `PingException` clause has a filter, wouldn't this wrap any `PingException`'s that weren't due to a cancellation in a second outer `PingException`?

Reply:
Good catch. I copied the existing code without carefully thinking about it. I think we don't need to catch all exception here. Will remove it.